### ISSUE TITLE
BigQuery: Create table external_data_configuration ignoring some settings

### ISSUE THREAD

**Issue by QuinRiva** [googleapis/google-cloud-python#9821]
#### Environment details
2. OS type and version: Centos 7
3. Python version: 3.6.9
4. google-cloud-bigquery: version 1.20

#### Steps to reproduce

  1. Create a table using a Google Sheet as the data source

#### Code example

```python
schema = [
        bigquery.SchemaField("id", "INTEGER", mode="NULLABLE"),
        bigquery.SchemaField("name", "INTEGER", mode="NULLABLE")
    ]
tbx = bigquery.Table(dataset.table('tblName'), schema=schema)
bq_client.delete_table(tbx, not_found_ok=True)
external_config = bigquery.ExternalConfig("GOOGLE_SHEETS")
sheet_url = ("https://docs.google.com/spreadsheets/sheetURI")
external_config.source_uris = [sheet_url]
external_config.autodetect = True
external_config.ignoreUnknownValues = True
external_config.maxBadRecords = 2
external_config.options.skip_leading_rows = 1
external_config.options.range = ('sheetName!A:B')
tbx.external_data_configuration = external_config
tbl = bq_client.create_table(tbx)
```

The table is created, however the following options are ignored:
- autodetect
- ignoreUnkwonValues
- maxBadRecords
- range

It appears that only the following are actually passed through:
- source_uris
- source format
- skip_leading_rows

If the table is create manually through the cloud console UI, all of the aforementioned options work fine.

**Comment by HemangChothani** [comment]
@QuinRiva Everything works fine for me,here is the sample code and sample table data which i have used.
![Screenshot from 2019-11-21 17-25-50](https://user-images.githubusercontent.com/50404902/69336658-09092780-0c85-11ea-8b01-b73c0ad92cfe.png)






    client = bigquery.Client(credentials=credentials, project=project)
    dataset = client.get_dataset(dataset_id)
    table_id = "table_id"
    schema = [
        bigquery.SchemaField("name", "STRING"),
        bigquery.SchemaField("post_abbr", "STRING"),
        bigquery.SchemaField("bad_record", "INTEGER"),
    ]

    table = bigquery.Table(dataset.table(table_id), schema=schema)
    client.delete_table(table, not_found_ok=True)
    external_config = bigquery.ExternalConfig("GOOGLE_SHEETS")
     
    
    external_config.source_uris = ["sheet_url"]
    external_config.options.skip_leading_rows = 1  
    external_config.options.range = ("sheetName!A1:C6")

    # if `schema=None` it works fine.
    external_config.autodetect = True   

    # if the value is `1` then it throws an error as in the table cell `C3` and `C5` has character. 
    external_config.max_bad_records = 2    

    # Verify with the csv and json it works (through the error if it's value is False and csv has an extra column).
    external_config.ignore_unknown_values = True   
    table.external_data_configuration = external_config

    table = client.create_table(table)

   
    sql = "SELECT * FROM `{}.{}`".format(dataset_id, table_id)
    query_job = client.query(sql)  

    states = list(query_job)
    print (states)

**Comment by QuinRiva** [comment]
Can you try with a range that isn't on sheet 1? Targeting others sheets write with the range command is what I've found to be most problematic.

**Comment by HemangChothani** [comment]
@QuinRiva  I have tried range function with sheet2  and sheet3 like `sheet2!A1:C6` and `sheet3!A1:C6` it's working fine for me, but i think it only works with single sheet at a time, doesn't work with multiple like `sheet2!A1:sheet3!C6`.
if it's not adequate can you please share a problem or error with sample code?

**Comment by QuinRiva** [comment]
Hmm, I spent a day trying to get it working and couldn't  - I'll try to spin something up in a Colab and share it.

**Comment by HemangChothani** [comment]
@QuinRiva  are you satisfied with the answer(solution) , may i close the issue?

**Comment by QuinRiva** [comment]
You are correct.  I rewrote my code from scratch - and it worked.  I must have had a typo somewhere.

**Comment by HemangChothani** [comment]
Please feel free to re-open this issue if any doubt or it restate.

**Comment by etheleon** [comment]
The ability to specify the range only was included >= 1.21.0
